test/topology/test_topology.py

```diff
@@ -112,6 +114,51 @@ async def test_replace_reuse_ip(manager: ManagerClient, random_tables) -> None:
     # TODO: check that group 0 no longer contains the replaced node (#12153)
 
 
+async def reconnect_driver(manager: ManagerClient) -> Session:
+    """Workaround for scylladb/python-driver#170:
+       the existing driver session may not reconnect, create a new one.
+    """
+    logging.info(f"Reconnecting driver")
+    manager.driver_close()
+    await manager.driver_connect()
+
+# Checks basic functionality on the cluster with different values of the --smp parameter on the nodes.
+@pytest.mark.asyncio
+async def test_nodes_with_different_smp(manager: ManagerClient, random_tables: RandomTables) -> None:
+    # In this test it's more convenient to start with a fresh cluster.
+    # We don't need the default nodes,
+    # but there is currently no way to express this in the test infrastructure
+    servers = await manager.running_servers()
+    await manager.cql.run_async(f"DROP KEYSPACE {random_tables.keyspace}")
+    await asyncio.gather(*(manager.server_stop_gracefully(s.server_id) for s in servers))
+
+    # When the node starts it tries to communicate with others
+    # by sending group0_peer_exchange message to them.
+    # This message can be handled on arbitrary shard of the target node.
+    # The method manager.server_add waits for node to start, which can only happen
+    # if this message has been handled correctly.
+    #
+    # Note: messaging_service is initialized with server_socket::load_balancing_algorithm::port
+    # policy, this means that the shard for message will be chosen as client_port % smp::count.
+    # The client port in turn is chosen as rand() * smp::count + current_shard
+    # (posix_socket_impl::find_port_and_connect).
+    # If this succeeds to occupy a free port in 5 tries and smp::count is the same
+    # on both nodes, then it's guaranteed that the message will be
+    # processed on the same shard as the calling code.
+    # In the general case, we cannot assume that this same shard guarantee holds.
+    await manager.server_add(cmdline=['--smp', '3'])
```

Comment:
How is this even possible, you're adding new nodes to a stopped cluster :O

Reply:
> The new leader should be elected.

You first have to add a node to the Raft configuration for it to be elected leader. You cannot add a node to Raft config without a quorum alive in the existing config. Here all existing nodes are dead. So you cannot even ask an existing node to start the reconfiguration procedure.

Even with Raft disabled, this should be impossible, a bootstrapping node needs to contact an existing node to bootstrap.

Reply:
This is a new, unrelated cluster, since `_seeds` in `ScyllaCluster` ignores stopped nodes

Reply:
Ugh